### ISSUE TITLE
Use keycloak as Dex OIDC provider

### ISSUE THREAD

**Issue by jwim5819** [kubevela/velaux#689]
I'm trying to connect dex oidc provider as keycloak
here's my dex config
![1](https://user-images.githubusercontent.com/106866750/223615588-852ecc5d-9b2a-4cdd-b26a-89d8c6012fc1.png)
after setting everything up.
It looks like a success to login.
![login_page_1](https://user-images.githubusercontent.com/106866750/223615714-61850b1b-5286-4c11-b5cd-d2b54323120c.png)
![login_page_2](https://user-images.githubusercontent.com/106866750/223615720-010fa669-b2aa-4cce-90a8-5b44be9261bb.png)
![login_page_3](https://user-images.githubusercontent.com/106866750/223615730-2bb4f6cb-250c-4298-8c37-0e15db59e35a.png)
![2](https://user-images.githubusercontent.com/106866750/223615743-0d432c5c-65fd-43bd-8baf-b81d3d7aae3d.png)
but "entity index is invalid(500)" error occurred

Is there any solutions?

**KubeVela Version**
1.7.5

**Comment by barnettZQG** [comment]
@jwim5819 Can you check the logs of the API server?

```bash
vela logs -n vela-system addon-velaux -c apiserver
```

**Comment by jwim5819** [comment]
here's API server Pod log and Dex Pod log
![apiserver](https://user-images.githubusercontent.com/106866750/223662324-8960dfde-7323-4b38-9d34-c66d5fa14909.png)
I0308 08:28:40.315542 1 server.go:271] "request log" clientIP="192.168.174.1" path="/api/v1/auth/login_type" method="GET" status=200 time="6.813222ms" responseSize=23                                                                                                                                         
I0308 08:28:40.333227 1 server.go:271] "request log" clientIP="192.168.174.1" path="/api/v1/auth/dex_config" method="GET" status=200 time="2.964487ms" responseSize=155                                                                                                                                        
E0308 08:28:45.420848 1 bcode.go:111] Business exceptions, error message: entity index is invalid, path:/api/v1/auth/login method:POST
I0308 08:28:45.420970 1 server.go:271] "request log" clientIP="192.168.174.1" path="/api/v1/auth/login" method="POST" status=500 time="47.071691ms" responseSize=63

![dex](https://user-images.githubusercontent.com/106866750/223662678-332474d0-35e1-4309-9bc9-fd5fb8d29d17.png)

**Comment by barnettZQG** [comment]
@jwim5819 Maybe your user email or name includes invalid symbols to lead to this issue. Do you test with other accounts?

**Comment by jwim5819** [comment]
yes i tested Ordinary account but it didn't work either

**Comment by jwim5819** [comment]
i solved it.
It happened because there was no db in velaUX

**Comment by wonderflow** [comment]
@jwim5819 Can you provide the steps about how you solved it here?

**Comment by barnettZQG** [comment]
@jwim5819 You are right, this issue only occurs when using the kube-api as the database.

**Comment by lucasfcnunes** [comment]
So, can't use kube-api (configmaps and secrets) to config the app?
